cypress/e2e/RoundTrips/TextEditor.feature

```diff
@@ -0,0 +1,605 @@
+    And  I click on '[data-cy="file-explorer"] [data-cy="folder-button_{{ diagramFolder }}/folder"]'
+    Then I expect '[data-cy="file-explorer-action-menu"] [data-cy="create-file-action-item"]' exists
+
+    When I click on '[data-cy="file-explorer-action-menu"] [data-cy="create-file-action-item"]'
+    Then I expect '[data-cy="create-file-dialog"]' exists
+
+    When I set on '[data-cy="create-file-form"] [data-cy="name-input"]' text 'file.js'
+    And  I click on '[data-cy="create-file-form"] [data-cy="submit-button"]'
+    Then I expect 'positive' toast to appear with text 'Folder is created &#129395;!'
+    And  I expect '[data-cy="create-file-form"]' is closed
+    And  I expect '[data-cy="file-tabs-container"] [data-cy="file_{{ diagramFolder }}/main.tf"]' exists
+    And  I expect '[data-cy="file-tabs-container"] [data-cy="file_newFile"]' exists
+    And  I expect '[data-cy="file-tabs-container"] [data-cy="file_{{ diagramFolder }}/folder/file.js"]' exists
+    And  I expect '[data-cy="file-tabs-container"] [data-cy="active-tab"]' is 'file.js'
+
+    # Delete diagramFolder
+    When I click on '[data-cy="file-explorer"] [data-cy="folder-button_{{ diagramFolder }}"]'
+    Then I expect '[data-cy="file-explorer-action-menu"]' exists
+
+    When I click on '[data-cy="file-explorer-action-menu"] [data-cy="delete-file-action-item"]'
+    Then I expect '[data-cy="delete-file-dialog"]' exists
+    And  I expect '[data-cy="delete-file-form"] [data-cy="confirm-delete-checkbox"]' exists
+    And  I expect checkbox '[data-cy="delete-file-form"] [data-cy="confirm-delete-checkbox"]' is not checked
+    And  I expect '[data-cy="delete-file-form"] [data-cy="submit-button"]' to be disabled
+
+    When I click on '[data-cy="delete-file-form"] [data-cy="confirm-delete-checkbox"]'
+    Then I expect checkbox '[data-cy="delete-file-form"] [data-cy="confirm-delete-checkbox"]' is checked
+
+    # Check all folders, files and corresponding tabs are deleted and closed
+    When I click on '[data-cy="delete-file-form"] [data-cy="submit-button"]'
+    Then I expect 'positive' toast to appear with text 'Folder is deleted.'
+    And  I expect '[data-cy="delete-file-form"]' is closed
+    And  I expect '[data-cy="folder_{{ diagramFolder }}"]' not exists
+    And  I expect '[data-cy="folder_{{ diagramFolder }}/folder"]' not exists
+    And  I expect '[data-cy="file_{{ diagramFolder }}/folder"]' not exists
+    And  I expect '[data-cy="file_{{ diagramFolder }}/main.tf"]' not exists
+    And  I expect '[data-cy="file_{{ diagramFolder }}/nestedFile.js"]' not exists
+    And  I expect '[data-cy="file_{{ diagramFolder }}/folder/file.js"]' not exists
+    But  I expect '[data-cy="file-tabs-container"] [data-cy="active-tab"]' is 'newFile'
+
+    ## 609 Save the component in defaultFileName after deleting the diagram's file
+    # Back to the models page
+    When I visit the 'localhost:8080/#/projects/{{ projectName }}/models'
```

Comment:
why do you go to the model page by url when you can click on the project ?
BTW by clicking it will also make the "application is loaded" step useless (so you will be able to remove it)

Reply:
I forgot we have access to the button from the Text view 🫣